napalm_ios/ios.py

```diff
@@ -118,7 +120,7 @@ def __init__(self, hostname, username, password, timeout=60, optional_args=None)
             except KeyError:
                 pass
         self.global_delay_factor = optional_args.get('global_delay_factor', 1)
-        self.port = optional_args.get('port', 22)
+        self.port = optional_args.get('port', {'ssh': 22, 'telnet': 23}[self.transport])
```

Comment:
This and line 79 are in a separate PR...shouldn't them here (just going to cause issues).